### ISSUE TITLE
Uchiwa breaks Sensu API on non-existent client URL request

### ISSUE THREAD

**Issue by mdzidic** [sensu/uchiwa#815]
## Expected Behavior
It should just return 404 message.

## Current Behavior
If You open Uchiwa dashboard and request details for non-existent client after few seconds it will throw error:
`Could not retrieve stashes from datacenter, No healthy APIs available for datacenter`

## Possible Solution
I got information on Slack that Uchiwa (as of 1.6.0) performs health checks against /info API endpoint on each configured host. if /info is returning non-OK responses for a given API host, the host will be marked as unhealthy and wont be used until it appears healthy again.

I couldn't reproduce this error on Uchiwa 1.5.0

## Steps to Reproduce (for bugs)

1. Open Uchiwa dashboard with non-existent client in URL eg. https://uchiwa:3000/#/client/{Your-Sensu-DC}/unknown-client-id
2. Wait for a few seconds (~ 10s) and Uchiwa will start throwing error like `Could not retrieve stashes from datacenter, No healthy APIs available for datacenter`
3. API will recover only if You leave page with non-existent client.

## Context
Uchiwa log:
```
{"timestamp":"2019-03-28T16:42:10.448265379Z","level":"warn","message":"404 Not Found"}
{"timestamp":"2019-03-28T16:42:10.449883174Z","level":"warn","message":"No healthy APIs available for datacenter: Sensu"}
{"timestamp":"2019-03-28T16:42:15.389876423Z","level":"warn","message":"could not retrieve clients entries from datacenter Sensu: No healthy APIs available for datacenter: Sensu"}
{"timestamp":"2019-03-28T16:42:15.389931196Z","level":"warn","message":"failed to update the datacenter Sensu in 112.023µs"}
{"timestamp":"2019-03-28T16:42:15.389948375Z","level":"warn","message":"stopping aggregates"}
{"timestamp":"2019-03-28T16:42:15.389965638Z","level":"warn","message":"sensu api is unhealthy: http://sensu-api:4567 (datacenter: Sensu)"}
{"timestamp":"2019-03-28T16:42:20.442617545Z","level":"warn","message":"No healthy APIs available for datacenter: Sensu"}
{"timestamp":"2019-03-28T16:42:20.446438639Z","level":"warn","message":"No healthy APIs available for datacenter: Sensu"}
{"timestamp":"2019-03-28T16:42:25.389809968Z","level":"warn","message":"could not retrieve stashes from datacenter Sensu: No healthy APIs available for datacenter: Sensu"}
{"timestamp":"2019-03-28T16:42:25.389891472Z","level":"warn","message":"failed to update the datacenter Sensu in 185.257µs"}
{"timestamp":"2019-03-28T16:42:25.390243239Z","level":"warn","message":"checking health of sensu api http://sensu-api:4567 (datacenter: Sensu)"}
{"timestamp":"2019-03-28T16:42:25.396295793Z","level":"warn","message":"sensu api is healthy again: http://sensu-api:4567 (datacenter: Sensu)"}
```

## Your Environment
* Uchiwa version used: 1.6.0
* Sensu version used: 1.7.0
* Operating System and version (e.g. Ubuntu 14.04): CentOS 7

**Comment by cwjohnston** [comment]
@mdzidic thanks for opening this issue. Can you please provide Sensu API logs which correspond to the Uchiwa warnings shown in your report?

**Comment by toote** [comment]
I am running into the same issue, the default configuration of the API shows no logs whatsoever. I just see 2 requests for `/clients/CLIENT_NAME?dc=default` and `/clients/CLIENT_NAME/history?dc=default` that return a 404 every single time because indeed they have been removed

**Comment by mdzidic** [comment]
@cwjohnston I can't find anything inside Sensu API logs...

**Comment by oboukili** [comment]
I can confirm this happened recently for me as well, downgrade Uchiwa to 1.5.0 resolved it.

**Comment by annaplotkin** [comment]
@mdzidic Thanks so much for reporting this issue. Our team has triaged this and will pursue an update with a resolution by early next week. We appreciate your patience!

**Comment by mylesw42** [comment]
The problem still persists with for me on Uchiwa 1.5.0 & Sensu 1.7.0-2. Any update on when this fix may be available? It causes our Uchiwa cluster to flap constantly until we find out which user has left a browser window open on non-existent client. Thanks.

BTW, Here is the output that is produced by the sensu-api.log for a 404 client...

{"timestamp":"2019-05-13T15:27:46.640921-0600","level":"info","message":"api response","request":{"request_id":"511f1a8f-aa99-4fc4-b65c-c38f63f139c6","remote_address":"10.100.36.52","user_agent":"Go-http-client/1.1","method":"GET","uri":"/clients/<client name>","query_string":null,"body":""},"status":404,"content_length":0,"time":0.0}

**Comment by mylesw42** [comment]
My bad, I guess my version of Uchiwa is at 1.6.0, I double checked the installed RPM. I had mistakenly seen version 1.5.0 (cached) on the Uchiwa info page. I force refreshed and it's 1.6.0 on the info page now.

**Comment by cwjohnston** [comment]
Hello @mdzidic @toote @oboukili @mylesw42 

We've recently released [Uchiwa 1.7.0](https://github.com/sensu/uchiwa/blob/master/CHANGELOG.md#170-2019-05-22) containing the changes in https://github.com/sensu/uchiwa/pull/817.  We believe these changes will resolve the behavior described in this issue. 

At your earliest convenience, would you please test this latest release and share with us your feedback as to whether this issue is resolved?

**Comment by toote** [comment]
@cwjohnston I can confirm that the latest release has fixed this issue
